pkg/metrics/tls.go

```diff
@@ -181,10 +182,13 @@ func (t *tlsConfigurerImpl) updateClientCA(cm *v1.ConfigMap) {
 		}
 		var signerCAs []*x509.Certificate
 		for _, c := range certs {
-			if c.Issuer.CommonName == signerName {
+			if strings.HasPrefix(c.Subject.CommonName, signerName) {
```

Comment:
I also wonder how stable that subject name is. Thoughts on just increasing the client CA scope to all certificates in `extension-apiserver-authentication`?

Reply:
Hmm, I think the argument could be made for both sides regarding whether to consider this as stable.

It could be sufficient for now to verify the certificate itself against the cert pool created out of the CA of `extension-apiserver-authentication` (incl. the verification for the key usage), looking at the code of Kubernetes we are essentially following [the same principles there](https://github.com/kubernetes/apiserver/blob/master/pkg/server/dynamiccertificates/tlsconfig.go#L170-L187) (and looking at [the documentation for client auth within the api server](https://github.com/kubernetes-sigs/apiserver-builder-alpha/blob/master/docs/concepts/auth.md#client-certificate-authentication) that seems to be also somewhat in line).

Within the config map, will all certificates signed there be issued by the `kube-csr-signer`? If that's the case, then the point of doing the extra verification here is moot, since virtually all certificates contained within the map could be used to authZ against this endpoint (even if we may not intend them to).

Reply:
> Within the config map, will all certificates signed there be issued by the kube-csr-signer?

No, there are different issuers, for example `admin-kubeconfig-signer` and `kube-control-plane-signer`.

```
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number: 7874177071138344378 (0x6d46b253b7ba0dba)
    Signature Algorithm: sha256WithRSAEncryption
        Issuer: OU=openshift, CN=kube-control-plane-signer
        Validity
            Not Before: Jul  3 12:50:05 2023 GMT
            Not After : Jul  2 12:50:05 2024 GMT
        Subject: OU=openshift, CN=kube-control-plane-signer
```

Reply:
But it does sound like the expected behavior is to simply verify against the whole cert pool, even if that extends the allowed certificates. So I would probably opt to remove this check again and load all CA certs.